src/lib/udmp-parser.h

```diff
@@ -987,6 +1052,32 @@ class UserDumpParser {
     return ForegroundThreadId_;
   }
 
+  std::string to_string() const {
+    std::stringstream ss;
+    ss << "UserDumpParser(";
+    ss << "ModuleNb=" << Modules_.size();
+    ss << ", ThreadNb=" << Threads_.size();
+    ss << ")";
+    return ss.str();
+  }
+
+  std::vector<uint8_t> ReadMemory(uintptr_t Address, size_t Size) const {
+    std::vector<uint8_t> Out;
+
+    auto const Block = GetMemBlock(Address);
+    if (!Block) {
+      throw ParsingError(std::string("Invalid address"));
```

Comment:
If this is the only place we are using this exception; maybe we can make this function exceptionless?

Could simply return a `std::optional<std::vector<uint8_t>>` and returns `{}` in that case?

Reply:
Otherwise you should be able to simplify `ParsingError(std::string(""))` by `ParsingError("")`? Can also move `Out` after the block closer from where it is used

Reply:
I'm happy to change: I usually like failing loudly (hence exceptions) for parsing failures, or unexpected values. I realize they are expensive but they should (normally) not happen often. Also here, we can argue that a user can ask for a `0` size read access, in which case we can't differenciate between a read failure and just an empty result. But I'm anything but inflexible on this, so tell me.

Reply:
I don't really mind about perf, etc. but certain codebases forbid exceptions entirely. Also, in the case you are describing I think you should just return an empty vector if the user is asking to read a 0 size and only return `{}` when an issue occurred?